core/src/main/java/io/questdb/cairo/PartitionBy.java

```diff
@@ -183,36 +180,101 @@ public static boolean isPartitioned(int partitionBy) {
                 case DAY:
-                    ee.put(DAY_FMT);
-                    break;
-                case WEEK:
-                    ee.put(WEEK_FMT);
+                    fmtMethod = DAY_FORMAT;
+                    fmtStr = DAY_PATTERN;
                     break;
                 case MONTH:
-                    ee.put(MONTH_FMT);
+                    fmtMethod = MONTH_FORMAT;
+                    fmtStr = MONTH_PATTERN;
                     break;
                 case YEAR:
-                    ee.put(YEAR_FMT);
+                    fmtMethod = YEAR_FORMAT;
+                    fmtStr = YEAR_PATTERN;
                     break;
                 case HOUR:
-                    ee.put(HOUR_FMT);
+                    fmtMethod = HOUR_FORMAT;
+                    fmtStr = HOUR_PATTERN;
+                    break;
+                case WEEK:
+                    fmtMethod = WEEK_FORMAT;
+                    fmtStr = WEEK_PATTERN;
+                    break;
+                case NONE:
+                    fmtMethod = DEFAULT_FORMAT;
+                    fmtStr = partitionName;
                     break;
+                default:
+                    throw new UnsupportedOperationException("partition by " + partitionBy + " does not have date format");
+            }
+            limit = fmtStr.length();
+            if (partitionName.length() < limit) {
+                throw expectedPartitionDirNameFormatCairoException(partitionName, partitionName.length(), partitionBy);
             }
-            ee.put("' expected");
-            throw ee;
+            return fmtMethod.parse(partitionName, 0, limit, null);
+        } catch (NumericException e) {
+            if (partitionBy == PartitionBy.WEEK) {
+                // maybe the user used a timestamp, or a date, string.
+                int local_limit = DAY_PATTERN.length();
+                long timestamp;
+                try {
+                    // trim to lowest precision needed and get the timestamp
+                    timestamp = DAY_FORMAT.parse(partitionName, 0, local_limit, null);
+                } catch (NumericException ignore) {
+                    throw expectedPartitionDirNameFormatCairoException(partitionName, Math.min(partitionName.length(), local_limit), partitionBy);
+                }
+
+                // convert timestamp to first day of the week
+                int year = Timestamps.getYear(timestamp);
+                int week = Timestamps.getWeek(timestamp);
+                if (week == 52 && Timestamps.getMonthOfYear(timestamp) == 1) {
+                    year--;
+                }
+                StringSink sink = Misc.getThreadLocalBuilder();
+                sink.put(year).put("-W");
+                if (week < 10) {
+                    sink.put('0');
+                }
+                sink.put(week);
+                local_limit = sink.length();
+                try {
+                    return WEEK_FORMAT.parse(sink, 0, local_limit, null);
```

Comment:
Do we really need this double parse dance with additional sink? Would be nice to avoid it and floor the first parsed timestamp directly.

Reply:
agreed, will simplify and make it faster too, good catch.

Reply:
done in the last two commits